### PR TITLE
Modrs 10

### PR DESCRIPTION
**Pull request by alekGbuz**
## Pre-Merge Checklist:
 Before merging this PR, please go through the following list and take appropriate actions.

 - Does this PR meet or exceed the expected quality standards?
   - [ ] Code coverage on new code is 80% or greater
   - [ ] Duplications on new code is 3% or less
   - [ ] There are no major code smells or security issues
 - Does this introduce breaking changes?
   - [ ] Were any API paths or methods changed, added or removed?
   - [ ] Were there any schema changes?
   - [ ] Did any of the interface versions change?
   - [ ] Were permissions changed, added, or removed?
   - [ ] Are there new interface dependencies?
   - [ ] There are no breaking changes in this PR.

 If there are breaking changes, please **STOP** and consider the following:

 - What other modules will these changes impact?
 - Do JIRAs exist to update the impacted modules?
   - [ ] If not, please create them
   - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
   - [ ] Do they have all they appropriate links to blocked/related issues?
 - Are the JIRAs under active development?
   - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
 - Do PRs exist for these changes?
   - [ ] If so, have they been approved?

 Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

 While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Review by siarhei-charniak — APPROVED** [review]
@alekGbuz your PR looks good, please take a look at review comments.

File: src/test/resources/mappings/users.json [hunk 1]
```diff
@@ -52,6 +52,19 @@
         }
       }
     },
+    {
+      "request": {
+        "method": "GET",
+        "url": "/users?query=id==21932a85-bd00-446b-9565-46e0c1a5490b"
```
Comment (khandramai) [inline src/test/resources/mappings/users.json:58]: x-okapi-token/x-okapi-tenant will be useful here just to be sure that correct values were populated during FeignClient processing.

File: src/test/resources/mappings/circulation.json [hunk 2]
```diff
@@ -11,6 +11,19 @@
           "Content-Type": "application/json"
         }
       }
+    },
+    {
+      "request": {
+        "method": "GET",
+        "url": "/circulation/requests/queue/0b96a642-5e7f-452d-9cae-9cee66c9a892"
```
Comment (khandramai) [inline src/test/resources/mappings/circulation.json:18]: x-okapi-token/x-okapi-tenant will be useful here just to be sure that correct values were populated during FeignClient processing.

File: src/test/resources/mappings/inventory_items.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "mappings": [
+    {
+      "request": {
+        "method": "GET",
+        "url": "/inventory/items?query=barcode==645398607547"
```
Comment (khandramai) [inline src/test/resources/mappings/inventory_items.json:6]: x-okapi-token/x-okapi-tenant will be useful here just to be sure that correct values were populated during FeignClient processing.

File: src/test/java/org/folio/rs/service/ReturnItemServiceTest.java [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ReturnItemServiceTest {
```
Comment (khandramai) [inline src/test/java/org/folio/rs/service/ReturnItemServiceTest.java:32]: If we use Mockito why do we need to have WireMock mappings? Please, make choice. If you are going to use Mockito (as well as Wiremock) x-okapi-token/x-okapi-tenant verification is mandatory.

Comment (khandramai) [inline src/test/java/org/folio/rs/service/ReturnItemServiceTest.java:32]: Why we use two separate test classes for Return flow and returnItemService? Is it possible to have all of them in the one class just to correspond all other tests in the module?

File: src/main/java/org/folio/rs/service/ReturnItemService.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+            retrievalQueueRepository.save(buildRetrievalRecord(itemRequest, item, users.getResult().get(0), remoteStorageConfigurationId));
+        });
+      }
+    }
+    checkInItemService.checkInItemByBarcode(remoteStorageConfigurationId, checkInItem);
+    log.info("Return success for item with barcode " + checkInItem.getItemBarcode());
+    return itemReturnResponse;
+  }
+
+  private RetrievalQueueRecord buildRetrievalRecord(Request itemRequest, Item item, User patron, String remoteStorageId) {
```
Comment (khandramai) [inline src/main/java/org/folio/rs/service/ReturnItemService.java:74]: Personally me prefer to use Spring Converter<A, B> for these purposes and create separate converter component.

File: src/main/java/org/folio/rs/service/ReturnItemService.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  private static final String BARCODE_QUERY_PROPERTY = "barcode==";
+  private static final String USER_ID_QUERY_PROPERTY = "id==";
+
+  private final InventoryClient inventoryClient;
+  private final CirculationClient circulationClient;
+  private final UsersClient usersClient;
+  private final RetrievalQueueRepository retrievalQueueRepository;
+  private final CheckInItemService checkInItemService;
+
+  public ReturnItemResponse returnItem(String remoteStorageConfigurationId, CheckInItem checkInItem) {
```
Comment (khandramai) [inline src/main/java/org/folio/rs/service/ReturnItemService.java:42]: Could you please refactor this method, for example, introducing separate methods for logic parts. Current method complexity make difficult to go through business logic.

**Comment by alekGbuz** [comment]
image

**Review by khandramai — APPROVED** [review]

**Pull request merged**